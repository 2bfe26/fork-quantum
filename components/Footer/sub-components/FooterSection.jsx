import React from 'react';
import PropTypes from 'prop-types';
import styled, { ThemeProvider } from 'styled-components';
import { Row, Col } from '../../Grid';
import FooterLinkGroup from './FooterLinkGroup';

const SectionTitle = styled.div`
  color: ${props => props.theme.primaryColor};
  font-family: ${props => props.theme.fontFamily};
  font-size: ${props => props.theme.titleFontSize};
  font-weight: ${props => props.theme.titleFontWeight};
  position: relative;
  margin-bottom: 15px;
  &::after {
    background-color: ${props => props.theme.primaryColor};
    position: absolute;
    bottom: 7px;
    margin-left: 10px;
    content: "-";
    width: 95%;
    text-indent: -5000px;
    height: 2px;
 }
`;

class FooterSection extends React.Component {
  constructor(props) {
    super(props);
    this.data = props.data;
  }

  render() {
    return (
      <ThemeProvider theme={this.data.theme}>
        <Row no-gutters>
          <Col desktop={12}>
            {
              this.data.title != null &&
              <SectionTitle>
                {this.data.title}
              </SectionTitle>
            }
            <Row no-gutters>
              {
                this.props.data.linkGroups.map(linkGroup => (
                  <Col desktop={linkGroup.cols} key={linkGroup.id}>
                    <FooterLinkGroup
                      data={linkGroup}
                    />
                  </Col>
                ))
              }
            </Row>
          </Col>
        </Row>
      </ThemeProvider>
    );
  }
}

// propTypes
FooterSection.propTypes = { data: PropTypes.instanceOf(Object) };
FooterSection.defaultProps = { data: null };

export default FooterSection;
