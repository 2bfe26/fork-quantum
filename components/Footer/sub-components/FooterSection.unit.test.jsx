import React from 'react';
import renderer from 'react-test-renderer';
import FooterSection from './FooterSection';

let defaultProps;

beforeEach(() => {
  defaultProps = {
    data: {
      title: 'Candidatos',
      theme: {
        primaryColor: '#005da4',
        secondaryColor: '#666666',
        titleFontSize: '14px',
        titleFontWeight: 'bold',
        linkFontSize: '12px',
      },
      linkGroups: [{
        id: 1,
        sideSeparator: true,
        cols: 4,
        links: [{
          id: 1,
          title: true,
          text: 'Buscar vagas',
          href: null,
          target: null,
          img: null,
        }],
      }],
    },
  };
});

describe('Default props', () => {
  it('Should match the snapshot', () => {
    const footerSection = <FooterSection data={defaultProps.data} />;
    const tree = renderer.create(footerSection).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('When do not have title', () => {
  it('Should match the snapshot', () => {
    defaultProps.data.title = null;

    const footerComponent = <FooterSection data={defaultProps.data} />;
    const tree = renderer.create(footerComponent).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
