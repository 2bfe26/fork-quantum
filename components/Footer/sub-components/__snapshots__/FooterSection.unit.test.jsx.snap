// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default props Should match the snapshot 1`] = `
<div
  className="Row-s1166su-0 nMSYG"
>
  <div
    className="Col-bi16rv-0 cygcwq"
  >
    <div
      className="FooterSection__SectionTitle-k2tayx-0 djKMMD"
    >
      Candidatos
    </div>
    <div
      className="Row-s1166su-0 nMSYG"
    >
      <div
        className="Col-bi16rv-0 ggiJSh"
      >
        <ul
          className="FooterLinkGroup__LinkGroup-s2tx9sw-0 ccSfWN"
        >
          <li
            className="title FooterLinkGroup__LinkItem-s2tx9sw-1 jCTQSx"
            href={null}
          >
            <a
              href={null}
            >
              Buscar vagas
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`When do not have title Should match the snapshot 1`] = `
<div
  className="Row-s1166su-0 nMSYG"
>
  <div
    className="Col-bi16rv-0 cygcwq"
  >
    <div
      className="Row-s1166su-0 nMSYG"
    >
      <div
        className="Col-bi16rv-0 ggiJSh"
      >
        <ul
          className="FooterLinkGroup__LinkGroup-s2tx9sw-0 ccSfWN"
        >
          <li
            className="title FooterLinkGroup__LinkItem-s2tx9sw-1 jCTQSx"
            href={null}
          >
            <a
              href={null}
            >
              Buscar vagas
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
